opal/datatype/opal_datatype_cuda.c

```diff
@@ -111,6 +111,59 @@ bool  opal_cuda_check_one_buf(char *buf, opal_convertor_t *convertor )
     return ( ftable.gpu_is_gpu_buffer(buf, convertor));
 }
 
+/*
```

Comment:
The OPAL datatype packing and unpacking is too low level to provide support for memory allocation. I understand this seems to be a popular place to put CUDA functions, but I would suggest these functions move to opal/mca/common/cuda.